### DISCUSSION TITLE
Can not force importing face-api.esm.js

### DISCUSSION THREAD

**Discussion by sakhovski** [vladmandic/face-api#61] [Q&A] [answered]
Hello Vladimir,
found your fork of face-api two days ago. Kudos on work done! It is nice to have Typescript typings and sourcemaps. There is one small issue, which, as it seems, I can not get working. I can not force importing face-api.esm.js in my TS-class like this:
`import * as faceapi from '@vladmandic/face-api/dist/face-api.esm.js';`
With this import
`import * as faceapi from '@vladmandic/face-api/';`
it works perfectly in the browser (which imports face-api.esm.js), but breaks in my tests running in the Node (which imports face-api.node.js) with following Error:
`Cannot find module '@tensorflow/tfjs-node' from 'face-api.node.js'`

Did I do something wrong or some exports are missing? 

Best Regards,
Vladimir

**Comment by vladmandic** [comment, marked as answer]
> I can not force importing face-api.esm.js in my TS-class like this:
> `import * as faceapi from '@vladmandic/face-api/dist/face-api.esm.js';`

can you describe what happens in more details? that should work just fine  
especially since `import * as faceapi from '@vladmandic/face-api/';` simply points to the same file anyhow  

> breaks in my tests running in the Node (which imports face-api.node.js) with following Error:
> Cannot find module '@tensorflow/tfjs-node' from 'face-api.node.js'

that is normal and fully documented - `face-api.node.js`,  
see section 3.1 in the readme: <https://github.com/vladmandic/face-api#31-import-commonjs>

**Reply by sakhovski** [reply]
Turned out the real problem was in electron runner for jest, which was running with node version, which was different from the node version being used to compile tensorflow.

**Reply by vladmandic** [reply]
ahh, makes sense but not easy to spot immediately - glad its resolved.
